game/src/ecs/components/ai/transition/ProtectOnAttack.java

```diff
@@ -0,0 +1,66 @@
+package ecs.components.ai.transition;
+
+import ecs.components.HealthComponent;
+import ecs.components.MissingComponentException;
+import ecs.components.PlayableComponent;
+import ecs.entities.Entity;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Lets AI protect an entity
+ *
+ * <p>Implements an AI that protects a specific entity with a HealthComponent if the hero afflicting damage to it.
+ * Entity will stay in fight mode</p>
+ */
+public class ProtectOnAttack implements ITransition {
+
+    private boolean isInfight = false;
+
+    private final List<Entity> toProtect = new ArrayList<>();
+
+    /**
+     * Constructor for one entity to protect
+     *
+     * @param entity
+     */
+    ProtectOnAttack(Entity entity) {
+        if (entity.getComponent(HealthComponent.class).isEmpty()) {
+            throw (new MissingComponentException("HealthComponent"));
+        }
+
+        this.toProtect.add(entity);
+
+    }
+
+    /**
+     * Constructor for a list of entities to protect
+     *
+     * <p>Checks if HealthComponent isPresent and adds it to the list of entities to protect</p>
+     *
+     * @param entities - Entities that are protected
+     */
+    ProtectOnAttack(List<Entity> entities) {
```

Comment:
```suggestion
    ProtectOnAttack(Collection<Entity> entities) {
```

hier reicht ein generischer Typ